packages/contracts/contracts/optimistic-ethereum/OVM/oracle/OVM_L1Oracle.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.5.0 <0.8.0;
+pragma experimental ABIEncoderV2;
+
+/* Library Imports */
+import { Lib_OVMCodec } from "../../libraries/codec/Lib_OVMCodec.sol";
+
+/* Interface Imports */
+import { iOVM_CanonicalTransactionChain } from "../../../../contracts/optimistic-ethereum/iOVM/chain/iOVM_CanonicalTransactionChain.sol";
+import { iOVM_L1StandardBridge } from "../../../../contracts/optimistic-ethereum/iOVM/bridge/tokens/iOVM_L1StandardBridge.sol";
+import { iOVM_L1Oracle } from "../../../../contracts/optimistic-ethereum/iOVM/oracle/iOVM_L1Oracle.sol";
+import { iOVM_L1ClaimableERC721 } from "../../../../contracts/optimistic-ethereum/iOVM/oracle/iOVM_L1ClaimableERC721.sol";
+
+contract OVM_Oracle is iOVM_L1Oracle {
+
+    iOVM_L1StandardBridge public ovmL1StandardBridge;
+    iOVM_CanonicalTransactionChain public ovmCanonicalTransactionChain;
+    iOVM_L1ClaimableERC721 public ovmL1ClaimableERC721;
+
+    bytes4 private constant _INTERFACE_ID_FAST_WITHDRAW    = 0xfc2bb3a8; // fastWithdraw(address,uint256,uint256,uint256,uint256,uint32,bytes)
+    bytes4 private constant _INTERFACE_ID_FAST_WITHDRAW_TO = 0xdaaab375; // fastWithdrawTo(address,address,uint256,uint256,uint256,uint256,uint32,bytes)
+
+    function initialize (
+        address _ovmL1StandardBridge,
+        address _ovmCanonicalTransactionChain,
+        address _ovmL1ClaimableERC721
+    )
+        external
+    {
+        require(
+            _ovmL1StandardBridge          != address(0) &&
+            _ovmCanonicalTransactionChain != address(0) &&
+            _ovmL1ClaimableERC721         != address(0),
+            "Must provide contract address"
+        )
+        ovmL1StandardBridge          = iOVM_L1StandardBridge(_ovmL1StandardBridge);
+        ovmCanonicalTransactionChain = iOVM_CanonicalTransactionChain(_ovmCanonicalTransactionChain);
+        ovmL1ClaimableERC721         = iOVM_L1ClaimableERC721(_ovmL1ClaimableERC721);
+    }
+
+    function processFastWithdrawal (
+        Lib_OVMCodec.Transaction calldata _transaction,
```

Comment:
- Can the `memory` parameter receive the value of the `calldata` parameter?
- Is the member of the calldata structure also treated as calldata?

Reply:
if not, we should use Lib_BytesUtils.sol.